app/routes/titiler/algorithms.py

```diff
@@ -26,24 +46,78 @@ def __call__(self, img: ImageData) -> ImageData:
+        data_alert_confidence = data // 10000
+
+        alert_confidence_map: OrderedDict = OrderedDict(
+            {
+                DeforestationAlertConfidence.low: {
+                    "confidence": 2,
+                    "colors": {"red": 237, "green": 164, "blue": 194},
+                },
+                DeforestationAlertConfidence.high: {
+                    "confidence": 3,
+                    "colors": {"red": 220, "green": 102, "blue": 153},
+                },
+                DeforestationAlertConfidence.highest: {
+                    "confidence": 4,
+                    "colors": {"red": 201, "green": 42, "blue": 109},
+                },
+            }
+        )
 
-        # Will add this in next iteration
-        # if self.start_date:
-        #     start_mask = data % 10000 >= (
-        #         np.datetime64(self.start_date).astype(int) - self.START_DATE
-        #     )
-        #     mask = mask * start_mask
-        # if self.end_date:
-        #     end_mask = data % 10000 <= (
-        #         np.datetime64(self.end_date).astype(int) - self.START_DATE
-        #     )
-        #     mask = mask * end_mask
-
-        r = np.where(data > 0, 228, 0)
-        g = np.where(data > 0, 102, 0)
-        b = np.where(data > 0, 153, 0)
+        r = np.zeros_like(data_alert_confidence, dtype=int)
+        g = np.zeros_like(data_alert_confidence, dtype=int)
+        b = np.zeros_like(data_alert_confidence, dtype=int)
+
+        for confidence_level, properties in alert_confidence_map.items():
+            confidence = properties["confidence"]
+            colors = properties["colors"]
+
+            r = np.where(
+                data_alert_confidence >= confidence,
+                colors["red"],
+                r,
+            )
+            g = np.where(
+                data_alert_confidence >= confidence,
+                colors["green"],
+                g,
+            )
+            b = np.where(
+                data_alert_confidence >= confidence,
+                colors["blue"],
+                b,
+            )
 
         intensity = np.where(mask == 0, img.data[1], 0)
+        if self.start_date:
+            start_mask = alert_date >= (
+                np.datetime64(self.start_date).astype(int)
+                - np.datetime64(self.START_DATE).astype(int)
+            )
+        if self.end_date:
+            end_mask = alert_date <= (
+                np.datetime64(self.end_date).astype(int)
+                - np.datetime64(self.START_DATE).astype(int)
+            )
+
+        if self.alert_confidence:
+            confidence_mask = (
+                data_alert_confidence
+                >= alert_confidence_map[self.alert_confidence]["confidence"]
+            )
+
+        intensity = np.where(
+            (
+                (mask == 0)
+                & (start_mask.astype(int) == 1)
+                & (end_mask.astype(int) == 1)
+                & (confidence_mask.astype(int) == 1)
```

Comment:
**It seems like you're intending to support a case where the caller can set self.alert_confidence or the start/end date to None if they don't want to mask on that.  But I don't think that will work, because in that case, start_mask, end_mask, or confidence_mask will be uninitialized variables, based on the if checks above, right?